src/util.cc

```diff
@@ -313,12 +411,83 @@ void GetWin32EscapedString(const string& input, string* result) {
   result->push_back(kQuote);
 }
 
+#ifdef _WIN32
+bool UTF8ToWide(wstring* wide, const string& utf8) {
+  if (utf8.empty()) {
+    wide->clear();
+  } else {
+    int dw = MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS, utf8.c_str(), utf8.size(), NULL, 0);
+    if (dw == 0) {
+      return false;
+    }
+    wide->resize(dw);
+    if (MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS, utf8.c_str(), utf8.size(), &(*wide)[0], wide->size()) != dw) {
+      return false;
+    }
+  }
+  return true;
+}
+
+// can it be memoized?
+static bool GetCWD(wstring* result) {
+  DWORD dw = GetCurrentDirectoryW(result->size(), &(*result)[0]);
+  if (dw == 0) {
+    return false;
+  }
+  if (dw > result->size()) {
+    result->resize(dw);
+    if (GetCurrentDirectoryW(dw, &(*result)[0]) != dw-1) {
+      return false;
+    }
+    result->resize(dw-1);
+  } else {
+    result->resize(dw);
+  }
+  return true;
+}
+
+static bool IsPathAbsoluteW(const wstring& wpath) {
+  return wpath.length() >= 2 && ( (wpath[0] == '\\' && wpath[1] == '\\') ||
+    (('a' <= (wpath[0]|0x20) && (wpath[0]|0x20) <= 'z' && wpath[1] == ':')) );
+}
+
+bool FullPathName(wstring* fullpath, const string& path, string* err) {
+  if (!UTF8ToWide(fullpath, path)) {
```

Comment:
can we do this just before L472 and use non-wide function for path manipulation?

Reply:
It is possible, but GetCurrentDirectory must be wide.
We cannot use result of GetCurrentDirectoryA, because it is not UTF8.
So using non-wide function for path manipulation means we have to
convert wide argument and also convert the result back ro wide.
This way we anyway get the wide function for path manipulation, only by wrapping non-wide one with the converters